pkg/api/v1alpha1/cluster_types.go

```diff
@@ -870,6 +870,33 @@ const (
 	MachineInvalidReason FailureReasonType = "MachineInvalid"
 )
 
+// ClusterCNIState is a type for defining statuses for Cluster default cni.
+type ClusterCNIState string
+
+const (
+	// ClusterCNIInitializing reports that the default CNI is initializing.
+	ClusterCNIInitializing ClusterCNIState = "initializing"
+
+	// ClusterCNIInstalled reports that the default CNI has been installed.
+	ClusterCNIInstalled ClusterCNIState = "installed"
+
+	// ClusterCNIUpdating reports that the default CNI is updating.
+	ClusterCNIUpdating ClusterCNIState = "updating"
+)
+
+// ClusterCNI represents a CNI cluster component.
+type ClusterCNI struct {
+	// Name of the default CNI
+	Name string `json:"name,omitempty"`
+
+	// Empty if not applied or no version can be inferred.
+	// +optional
+	Version string `json:"version,omitempty"`
+
+	// State reports the current state of the default CNI.
+	State ClusterCNIState `json:"state,omitempty"`
```

Comment:
is this really optional?

Reply:
it's not supposed to be optional. are you referring to the omitempty tag? that can be removed